### PR TITLE
Add method to compute Eclipse Centroid for CpGrid

### PR DESCRIPTION
**Pull request by aritorto**
For CpGrid, a method to compute centers of elements as in an Eclipse Grid is added and tested.

### CODE REVIEW

**Comment by aritorto** [comment]
jenkins build this serial please

File: CMakeLists_files.cmake [hunk 1]
```diff
@@ -112,6 +112,7 @@ list (APPEND TEST_SOURCE_FILES
 if(Boost_VERSION_STRING VERSION_GREATER 1.53)
 	list(APPEND TEST_SOURCE_FILES
 	  tests/cpgrid/disjointPatches_test.cpp
+	  tests/cpgrid/eclCentroid_test.cpp
```
Comment (bska) [inline CMakeLists_files.cmake:115]: This file does not appear to be included in the PR.  Did you accidentally not `add` it to `git`?

Comment (aritorto) [reply CMakeLists_files.cmake:115]: Probably. I confused  two branches. Let's me add the file. Thanks!

File: opm/grid/cpgrid/CpGridData.cpp [hunk 2]
```diff
@@ -2263,5 +2263,56 @@ CpGridData::refinePatch(const std::array<int,3>& cells_per_dim, const std::array
 }
 
 
+std::array<double,3> CpGridData::getAverageArr(const std::vector<std::array<double,3>>& vec) const
+{
+    assert(!vec.empty());
+    std::array<double,3> result = {0., 0., 0.};
+    for (int coord = 0; coord < 3; ++coord) {
+        for (const auto& arr : vec) {
+            result[coord] += (arr.at(coord))/static_cast<double>(vec.size());
```
Comment (blattms) [inline opm/grid/cpgrid/CpGridData.cpp:2272]: I think we should make this as fast as possible and use `arr[coord]` as `at` will do some additional checks. I guess we can be sure here that the entries are there as we have put them there.

Another option would be to move the division outside of the loops. Not 100% sure whether that will be faster, though.

Comment (aritorto) [reply opm/grid/cpgrid/CpGridData.cpp:2272]: Thanks! at() is replaced by [ ] now.

**Comment by aritorto** [comment]
jenkins build this serial please

File: tests/cpgrid/eclCentroid_test.cpp [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+        4*5
+        /
+        ACTNUM
+        5*1
+        /
+        PORO
+        5*0.15
+        /)";
+
+    createEclGridCpGrid_and_checkEclCentroid(deckString);
```
Comment (bska) [inline tests/cpgrid/eclCentroid_test.cpp:142]: Would it be possible to additionally have some checks of the form
```C++
BOOST_CHECK_CLOSE(centroid_x, expected_value_x, tolerance);
BOOST_CHECK_CLOSE(centroid_y, expected_value_y, tolerance);
BOOST_CHECK_CLOSE(centroid_z, expected_value_z, tolerance);
```
The checker function "just" prints values to the standard output stream, which means that a person will have to review the output manually and we lose a little bit of the automated aspect of unit tests.

**Review by bska — COMMENTED** [review]
I think this is good.  I just noticed a couple of potential improvements.

File: tests/cpgrid/eclCentroid_test.cpp (hunk 3 above)
Comment (aritorto) [reply tests/cpgrid/eclCentroid_test.cpp:142]: The printing is not necessary, I agree. However, the old line 96 was doing the type of check you suggested. I added one extra line in that spirit  "centroid_*, expected_value_*, tolerance", and removed the unnecessary print-part. 
Thanks!

File: opm/grid/cpgrid/CpGridData.cpp [hunk 4]
```diff
@@ -2263,5 +2263,56 @@ CpGridData::refinePatch(const std::array<int,3>& cells_per_dim, const std::array
...
+        }
+    }
+    return result;
+}
+
+std::array<double,3> CpGridData::computeEclCentroid(const int& idx) const
```
[the remark this answers is not in the document]
Comment (aritorto) [reply opm/grid/cpgrid/CpGridData.cpp:2278]: done!

File: opm/grid/cpgrid/CpGridData.cpp [hunk 5]
```diff
@@ -2263,5 +2263,56 @@ CpGridData::refinePatch(const std::array<int,3>& cells_per_dim, const std::array
...
+    for (int cornIdx = 0; cornIdx < 8; ++cornIdx) {
+        cell_to_point[cornIdx] = (this-> geometry_.geomVector(std::integral_constant<int,3>())
+                                  -> get(cell_to_point_indices[cornIdx])).center();
+    }
+
+    assert(!cell_to_point.empty());
+    // Recall corners order: cell_to_point ~ {0,1,2,3,4,5,6,7}
+    //  Top face      6---7      Bottom face     2---3
+    //               /   /                      /   /
+    //              4---5                      0---1
```
[the remark this answers is not in the document]
Comment (aritorto) [reply opm/grid/cpgrid/CpGridData.cpp:2293]: I reduced documentation, following this remark.

**Comment by aritorto** [comment]
jenkins build this serial please

**Comment by blattms** [comment]
Seems like everything from the reviews has been addressed. Cool, thanks. Just one more Jenkins run.

**Comment by blattms** [comment]
jenkins build this please

**Comment by aritorto** [comment]
jenkins build this serial please

**Pull request merged**